### PR TITLE
provide `run_with(core)` interface

### LINKED ISSUE #14: design question: external reactor
Heya, newbie question here. I was wondering why you chose to bundle in the event loop _inside_ the lib instead of taking in an external event loop. I know this was impossible in `mio` (at least I tried and failed), but was kinda hoping it'd be possible with `tokio`.

E.g. instead of this:
```rust
fn main() {
    let url = "https://replicate.npmjs.com/_changes".to_string();
    let mut changes = ChangesStream::new(url);

    changes.on(|change| {
        io::stdout().write_all(&change).unwrap();
    });

    changes.run();
}
```

I figured it'd be neat to be able to do this:
```rust
fn main() {
  let url = "https://replicate.npmjs.com/_changes".to_string();
  let mut changes = ChangesStream::new(url);
  let mut core = reactor::Core::new()

  changes.on(|change| {
    // And now for example we could use the same
    // core to do all sorts of timer stuff in here.
    io::stdout().write_all(&change).unwrap();
  });

  // slightly assuming core is able to run multiple things
  // and not just a single program; it'd be kinda moot otherwise hah
  core.run(changes);
}
```
I don't know; coming from Node I think I'm just very used to having a single event loop onto which all IO is attached; I was wondering if you'd have thoughts on this. Thanks heaps! :sparkles:

### PR DESCRIPTION
**Pull request by ashleygwilliams**
fixes #14

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -149,15 +149,19 @@ impl ChangesStream {
     ///    changes.run();
     /// # }
     /// ```
-    pub fn run(mut self) {
+    pub fn run(mut self, mut lp: Option<tokio_core::reactor::Core>) {
```
Comment (steveklabnik) [inline src/lib.rs]: So, while technically this works, It's a bit annoying to use. The common pattern is to use two functions instead:

    pub fn run(mut self)
    pub fn run_with(mut self, mut lp: tokio_core::reactor::Core)

Comment (ashleygwilliams) [inline src/lib.rs]: oh yeah i think i agree good call

**Comment by ashleygwilliams** [comment]
@yoshuawuyts you still want this? if so i will merge

**Comment by yoshuawuyts** [comment]
Yeah, I think that'd be rad!

**Pull request merged**